### PR TITLE
Finish questions

### PR DESCRIPTION
**Pull request by Arthien**

### CODE REVIEW

File: token/lib/token.ex [hunk 1]
```diff
@@ -6,13 +6,14 @@ defmodule Token do
 
   ## Examples
 
-      iex> Token.token_valid?("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsiYXBpIl0sImV4cCI6MTUwOTcwNTcyOSwiaWF0IjoxNTA3Mjg2NTI5LCJpc3MiOiJob21lcG9saXNoLmNvbSIsImp0aSI6IjhiMjY2OWNmLWEwNzItNDE0Zi1iMjgwLTcxYTU0MTFlZjc4OSIsIm5iZiI6MTUwNzI4NjUyOCwic3ViIjoiRGFsdG9uIiwidHlwIjoiYWNjZXNzIn0.a-bWuUABLoVrDo0EkJ_RWxYCPjBqfecdKW1ScJ1tFQ8")
+      iex> Token.token_valid?("eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsiYXBpIl0sImV4cCI6MTUyNjkzMjE0MywiaWF0IjoxNTI0NTEyOTQzLCJpc3MiOiJob21lcG9saXNoLmNvbSIsImp0aSI6IjI3NzNhNWQ4LWI0NDctNGUxZC1iMzhmLWNhMmM5NDAyNDQzMiIsIm5iZiI6MTUyNDUxMjk0Miwic3ViIjoiRGFsdG9uIiwidHlwIjoicmVmcmVzaCJ9.n0CmCoaeemTPLZ3SuFKoWuJUbSSP3H_je6H6bUL2bMxVa2CcaMcSv0NZYmDDYXegQ2nTbHommOLwuX8AeiSdrw")
```
Comment (erikreedstrom) [inline token/lib/token.ex]: Interesting. Why the change to the token?

File: queries/lib/queries.ex [hunk 2]
```diff
@@ -22,6 +31,11 @@ defmodule Queries do
   Note: Try something other than `or` clauses for this query.
   """
   def medium_cities_or_hot_cities_or_cold_small_cities() do
-    :not_implemented
+    City
+    |> join(:inner, [c], w in Weather, c.id == w.city_id)
+    |> where([c],  c.population > 5_000 and c.population < 1_000_000)
+    |> or_where([c, w], w.temp_lo > 30)
```
Comment (erikreedstrom) [inline queries/lib/queries.ex]: Can you think of a way to do this without an "or" query? Why might we not want to use an "or" condition here?

File: token/lib/token.ex (hunk 1 above)
Comment (Arthien) [reply token/lib/token.ex]: I noticed the test token was first generated last year and has expired. I generated a new token for the doctest. Perhaps I misunderstood the question and was supposed to implement a token refresh first.

Comment (erikreedstrom) [reply token/lib/token.ex]: No worries; there's no right answer. Just interested in how you approached it.

File: queries/lib/queries.ex (hunk 2 above)
Comment (Arthien) [reply queries/lib/queries.ex]: I suppose it could be done by creating a `dynamic` query. Based the parameters that come in we can drop certain "or" clauses and generate the query on the fly. This would allow pagination to be implemented easily without having to worry about ignored fields.

Comment (Arthien) [reply queries/lib/queries.ex]: I just noticed there are some redundant clauses here. The query could be trimmed down to reuse the overlapping results.

Comment (Arthien) [reply queries/lib/queries.ex]: @erikreedstrom Following up.
I would return all cities with a population < 1_000_000 and any city that is `not` between 24c and 30c. This inverse would get rid of all or statements and could give better performance.
